Haefner_IPHROS_MOGA.m

```diff
@@ -0,0 +1,508 @@
+    
+        %%%%%%%%%%%%%% (hyperplane constraint variable)
+        P_f_array = [P_f_collection;P_f_collection2];
+        S_f_array = [S_f_collection;S_f_collection2];
+        Q_f_array = [Q_f_collection;Q_f_collection2];
+        LHS_SVM = -1e6;
+        for i = 1:length(P_f_array)
+            var = -1334.934 + (18.222*P_f_array(i)) - (282.748*S_f_array(i)) - (200*(Q_f_array(i)/(24*264.172)));
+            if var > LHS_SVM
+                LHS_SVM = var;
+            end
+        end
+        %%%%%%%%%%%%%%
+    
+        % Constraint Variables
+        rr_max = max(rr_collective); 
+        rr_min = min(rr_collective); 
+        Q_f_max = Q_f_input; 
+        Q_f_min = Q_f; 
+        Q_c_max = max([Q_c_collection;Q_c_collection2]); 
+        Q_c_min = min([Q_c_collection;Q_c_collection2]); 
+        Q_p_max = max([Q_p_collection;Q_p_collection2]); 
+        Q_p_min = max([Q_p_collection;Q_p_collection2]);
+    end
+end
+
+function E_swht = turbine(V_swht,h,rho_sw,g,eta_ht)    
+    E_swht_joules = rho_sw*V_swht*g*h*eta_ht; % joules/day
+    E_swht = E_swht_joules/(3.6e6); % kWh/day
+end
+
+function E_htRO = energy_recapture(V_oRO,h,eta_ROio,rho_oRO,g,eta_ht)    
+    E_htRO_joules = rho_oRO*V_oRO*g*h*eta_ht*eta_ROio; % joules/day
+    E_htRO = E_htRO_joules/(3.6e6); % kWh/day
+end
+
+function [rho_ht,S_ht,V_ht] = mixing(V_oRO,V_swht,rho_oRO,S_oRO,gamma_RO,eta_RO,rho_sw,S_sw)
+    rho_ht = ((rho_sw*(1-gamma_RO)) + (gamma_RO*rho_oRO*(1-eta_RO)))/(1-(eta_RO*gamma_RO)); % kg/m^3
+    S_ht = ((S_sw*rho_sw*(1-gamma_RO)) + (gamma_RO*S_oRO*rho_oRO*(1-eta_RO)))/((rho_sw*(1-gamma_RO)) + (gamma_RO*rho_oRO*(1-eta_RO))); % g/kg
+    V_ht = V_swht + V_oRO; % m^3/day
+end
+
+function E_er = energy_summation(E_rd,E_swht,E_htRO)
+    E_er = E_rd+E_swht+E_htRO; % kWh/day
+end
+
+function C = RO_cost(N_e1,N_e2,N_pv1,N_pv2,M_cost_e1,M_cost_e2)
+    C1 = N_pv1*(M_cost_e1 + ((N_e1-1)*M_cost_e2)); % $ (proxy)
+    C2 = N_pv2*(M_cost_e1 + ((N_e2-1)*M_cost_e2)); % $ (proxy)
+        
+    C = C1 + C2; % $ (proxy)
+end
+
+% Miscellaneous Functions
+function S_brine = salinity_from_osmotic(pi,rho_fw,T,R,MW_salt_avg)
+    % Get Temperature and freshwater density
+    t = T; % celsius
+    T = (T)+273; % kelvin (from celsius)
+    
+    % Osmotic Coefficient Correlation (good for S>10)
+    a1 = 8.9453233003*10^-1;
+    a2 = 4.1560737424*10^-4;
+    a3 = -4.6262121398*10^-6;
+    a4 = 2.2211195897*10^-11;
+    a5 = -1.1445456438*10^-4;
+    a6 = -1.4783462366*10^-6;
+    a7 = -1.3526263499*10^-11;
+    a8 = 7.0132355546*10^-6;
+    a9 = 5.6960486681*10^-8;
+    a10 = -2.8624032584*10^-10;
+    
+    syms S real % g/kg
+    
+    phi = a1+(a2*t)+(a3*(t^2))+(a4*(t^4))+(a5*S)+(a6*S*t)+(a7*S*(t^3))+...
+        (a8*(S^2))+(a9*t*(S^2))+(a10*(S^2)*(t^2));
+    
+    % Converting osmotic pressure units
+    pi = pi/145.038; % MPa (from psi)
+    
+    % Getting numerical value for S
+    eqn = pi == ((phi*R*T*rho_fw)/(10^6))*(1000/MW_salt_avg)*...
```

Comment:
Where does this equation come from? I might have just missed it in the paper but I didn't see it